torch/csrc/jit/import_source.cpp

```diff
@@ -105,74 +133,98 @@ struct SourceResolver : public Resolver {
   std::unordered_map<std::string, std::shared_ptr<SugaredValue>> env_;
 };
 
-// Helper that contains the state for a parsing a TorchScript source string.
-struct SourceImporter {
-  SourceImporter(
-      const std::string& src,
-      const std::vector<at::Tensor>& constant_table)
-      : parser_(src) {
-    const auto version = parseVersionNumber();
-    resolver_ = std::make_shared<SourceResolver>(version, constant_table);
-  }
-
-  Parser parser_;
-  ResolverPtr resolver_;
+static size_t parseVersionNumber(Lexer& L) {
+  auto range = L.cur().range;
+  auto name = L.expect(TK_IDENT).text();
+  L.expect('=');
+  std::string version_text = L.expect(TK_NUMBER).text();
+  L.expect(TK_NEWLINE);
+  auto version = Const::create(L.cur().range, version_text);
+  if (name != "op_version_set")
+    throw ErrorReport(range) << "expected an assignment to op_version_set";
+  if (!version.isIntegral())
+    throw ErrorReport(range)
+        << "expected an integral version but found " << version.text();
+  return size_t(version.asIntegral());
+}
 
-  size_t parseVersionNumber() {
-    auto& L = parser_.lexer();
-    auto range = L.cur().range;
-    auto name = L.expect(TK_IDENT).text();
-    L.expect('=');
-    std::string version_text = L.expect(TK_NUMBER).text();
+static std::set<std::string> parseImports(Lexer& L) {
```

Comment:
Why are we importing them in a different order than they are listed in the file!? Errors will be weird!

Reply:
I feel like the de-dup and sorting should happen before export, not on import.